### PR TITLE
V1.2.1 user confirmation prompt

### LINKED ISSUE #35: User confirmation prompt
As a user, I would like to have confirmation when I perform commands like deleting a student so that I have a better peace of mind when it comes to the more dangerous commands

- Delete student particulars
- Clear student list
- Exit app

### PR DESCRIPTION
**Pull request by marcustw**
Added user confirmation prompt that requests CLI input when a user is trying to Clear, Delete, or Edit.

### CODE REVIEW

**Comment by marcustw** [comment]
28/9/2020 - still lacking test cases.

File: src/main/java/seedu/address/logic/commands/ConfirmationCommand.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ */
+public class ConfirmationCommand extends Command {
+    public static final String MESSAGE_CONFIRMATION = "(yes/no)";
+    public static final String MESSAGE_CONFIRMATION_DELETE = "Delete %1$d? " + MESSAGE_CONFIRMATION;
+    public static final String MESSAGE_CONFIRMATION_EDIT = "Edit %1$d? " + MESSAGE_CONFIRMATION;
+    public static final String MESSAGE_CONFIRMATION_CLEAR = "Clear list? " + MESSAGE_CONFIRMATION;
+    public static final String ACCEPT_COMMAND_1 = "yes";
+    public static final String ACCEPT_COMMAND_2 = "y";
+    public static final String REJECT_COMMAND_1 = "no";
+    public static final String REJECT_COMMAND_2 = "n";
```
Comment (erisjacey) [inline src/main/java/seedu/address/logic/commands/ConfirmationCommand.java]: Perhaps these could be renamed to `ACCEPT_COMMAND_FULL`, `ACCEPT_COMMAND_SHORT`, `REJECT_COMMAND_FULL` and `REJECT_COMMAND_SHORT`?

**Review by erisjacey — CHANGES_REQUESTED** [review]
LGTM! Just one small nitpick.

Will review again once test cases have been added 😄

File: src/main/java/seedu/address/logic/commands/ConfirmationRejectCommand.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.model.Model;
+
+/**
+ * Rejects the execution of a Dangerous Command.
```
Comment (nweiyue) [inline src/main/java/seedu/address/logic/commands/ConfirmationRejectCommand.java]: Perhaps the space between "Dangerous" and "Command" can be removed?

**Review by nweiyue — COMMENTED** [review]
LGTM with very extensive testing. Just a very small nitpick. Will review again when u finish ConfimationParserTest.

File: src/test/java/seedu/address/logic/commands/ConfirmCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+        assertTrue(confirmationDeleteCommand.equals(confirmationDeleteCommand));
+        assertTrue(confirmationEditCommand.equals(confirmationEditCommand));
+        assertTrue(confirmationClearCommand.equals(confirmationClearCommand));
+
+        assertFalse(confirmationDeleteCommand.equals(new ConfirmationCommand(secondDeleteCommand)));
+
+        assertFalse(confirmationDeleteCommand.equals(confirmationClearCommand));
+        assertFalse(confirmationDeleteCommand.equals(confirmationEditCommand));
+        assertFalse(confirmationEditCommand.equals(confirmationClearCommand));
```
Comment (erisjacey) [inline src/test/java/seedu/address/logic/commands/ConfirmCommandTest.java:46]: Perhaps comments could be added to explain what you're testing for each statement (+ expected result)?

File: src/test/java/seedu/address/logic/commands/ConfirmationAcceptCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        ConfirmationAcceptCommand confirmationAcceptEditCommand = new ConfirmationAcceptCommand(editCommand);
+        ConfirmationAcceptCommand confirmationAcceptClearCommand = new ConfirmationAcceptCommand(clearCommand);
+
+        assertTrue(confirmationAcceptDeleteCommand.equals(confirmationAcceptDeleteCommand));
+        assertTrue(confirmationAcceptEditCommand.equals(confirmationAcceptEditCommand));
+        assertTrue(confirmationAcceptClearCommand.equals(confirmationAcceptClearCommand));
+
+        assertFalse(confirmationAcceptDeleteCommand.equals(confirmationAcceptClearCommand));
+        assertFalse(confirmationAcceptDeleteCommand.equals(confirmationAcceptEditCommand));
+        assertFalse(confirmationAcceptEditCommand.equals(confirmationAcceptClearCommand));
```
Comment (erisjacey) [inline src/test/java/seedu/address/logic/commands/ConfirmationAcceptCommandTest.java]: I noticed a similar issue previously. Maybe more comments explaining each test case could be added?

**Review by erisjacey — CHANGES_REQUESTED** [review]
Left a few more comments! Now just waiting patiently for `ConfirmationParserTest.java` to be implemented. Great job so far :)

File: src/main/java/seedu/address/logic/commands/ConfirmationRejectCommand.java (hunk 2 above)
Comment (marcustw) [reply src/main/java/seedu/address/logic/commands/ConfirmationRejectCommand.java]: Thank you for the comment, I have resolved it!

File: src/test/java/seedu/address/logic/commands/ConfirmCommandTest.java (hunk 3 above)
Comment (marcustw) [reply src/test/java/seedu/address/logic/commands/ConfirmCommandTest.java:46]: Thank you for the comment, I have resolved it!

File: src/test/java/seedu/address/logic/commands/ConfirmationAcceptCommandTest.java (hunk 4 above)
Comment (marcustw) [reply src/test/java/seedu/address/logic/commands/ConfirmationAcceptCommandTest.java]: Thank you for the comment, I have resolved it!

**Review by marcustw — COMMENTED** [review]
I have added in ConfirmationParserTest with the edited comments and changes as recommended by reviewers.

**Review by erisjacey — APPROVED** [review]
LGTM 👍

**Pull request merged**